precise keyword throws error code

Adding 
```
float3x3 construct_ONB_Duff(float3 normal)
{
    precise float3x3 ret; // <---- precise
    // code here
    return ret;
}
```
to 

>DirectX-Graphics-Samples\Samples\Desktop\D3D12Raytracing\src\D3D12RaytracingRealTimeDenoisedAmbientOcclusion\RTAO\Shaders\RayGen.hlsli

causes:
> error MSB6006: "dxc.exe" exited with code -1073741819.

Seen w/ MSVC 2019, kit 10.0.18362.0

--- a/lib/HLSL/HLModule.cpp
+++ b/lib/HLSL/HLModule.cpp
@@ -1099,8 +1099,13 @@ void HLModule::MarkPreciseAttributeOnPtrWithFunctionCall(llvm::Value *Ptr,
           MarkPreciseAttributeOnValWithFunctionCall(arg, Builder, M);
         }
       } else {
-        IRBuilder<> Builder(CI->getNextNode());
-        MarkPreciseAttributeOnValWithFunctionCall(CI, Builder, M);
+        if (CI->getType()->isPointerTy()) {
+          // For instance, matrix subscript...
+          MarkPreciseAttributeOnPtrWithFunctionCall(CI, M);
+        } else {
+          IRBuilder<> Builder(CI->getNextNode());
+          MarkPreciseAttributeOnValWithFunctionCall(CI, Builder, M);
+        }
       }
     } else {
       // Must be GEP here.

--- a/tools/clang/test/HLSLFileCheck/hlsl/types/modifiers/precise/precise-matrix-subscript.hlsl
+++ b/tools/clang/test/HLSLFileCheck/hlsl/types/modifiers/precise/precise-matrix-subscript.hlsl
@@ -1,0 +1,12 @@
+// RUN: %dxc -E main -T vs_6_0 %s | FileCheck %s
+
+// CHECK: define void @main()
+
+float3x3 main(float3 normal : IN) : OUT
+{
+    precise float3x3 ret; // <---- precise
+    ret[0] = normal;
+    ret[1] = normal;
+    ret[2] = normal;
+    return ret;
+}

